### PR TITLE
sync send-to-helix.yml with runtime

### PR DESCRIPTION
**Pull request by ooooolivia**
Apply https://github.com/dotnet/runtime/commit/095628568dd6bc3236d1d1eaa1ac97fe9470152a changes in arcade
@trylek

### CODE REVIEW

**Review by trylek — APPROVED** [review]
Awesome, thanks so much for fixing this!

**Review by akoeplinger — APPROVED** [review]
Thank you!

File: eng/common/templates/steps/send-to-helix.yml [hunk 1]
```diff
@@ -30,40 +30,14 @@ parameters:
   continueOnError: false                 # optional -- determines whether to continue the build if the step errors; defaults to false
 
 steps:
-  - powershell: 'powershell "$env:BUILD_SOURCESDIRECTORY\eng\common\msbuild.ps1 $env:BUILD_SOURCESDIRECTORY\eng\common\helixpublish.proj /restore /t:Test /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\log\$env:BuildConfig\SendToHelix.binlog"'
+- template: /eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml
```
Comment (riarenas) [inline eng/common/templates/steps/send-to-helix.yml:33]: @ooooolivia This template is specific to runtime and is causing build breaks in arcade-validation. I'm going to revert this change.

Comment (riarenas) [reply eng/common/templates/steps/send-to-helix.yml:33]: Reverted in https://github.com/dotnet/arcade/pull/6230

Comment (trylek) [reply eng/common/templates/steps/send-to-helix.yml:33]: @riarenas - I must admit I'm getting more and more confused. Can you please explain to me why we needed to port these changes to arcade in the first place and why we can only port some of them? I guess there's some sharing going on between dotnet/arcade and dotnet/runtime w.r.t. the eng folder but it's apparently somehow filtered; what are the exact rules for this sharing?

Comment (riarenas) [reply eng/common/templates/steps/send-to-helix.yml:33]: The eng/common folder is shared across all repositories that use arcade. The rest of the eng folder is not. https://github.com/dotnet/arcade/blob/master/eng/common/README.md

Comment (trylek) [reply eng/common/templates/steps/send-to-helix.yml:33]: Hmm, does this then mean that the other change,

https://github.com/dotnet/arcade/pull/6195

is incorrect too? I mean, it's still not clear to me whether the sharing applies just to the <code>eng/common</code> folder or also its subfolders, but if it's the former, the quoted change is superfluous (because it's touching a non-shared location <code>eng/common/templates/steps/perf-send-to-helix.yml</code>), while if it's the latter, it is outright incorrect because it makes the shared YAML script call into a "non-shared" template <code>eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml</code>.

Comment (riarenas) [reply eng/common/templates/steps/send-to-helix.yml:33]: It's anything under eng/common/*

The original change seems incorrect to me for the reasons you pointed out. A common template is importing a non-common template, and that's what caused the break.

Comment (trylek) [reply eng/common/templates/steps/send-to-helix.yml:33]: OK, thanks for clarifying. While I understand the reasoning behind why the change is incorrect or incomplete, I still believe there's value in the code deduplication. Do you think we could get this resolved by moving the script <code>send-to-helix-inner-step.yml</code> to the shared <code>eng/common</code> subtree too, perhaps to the folder <code>eng/common/templates/steps</code> where the <code>perf-send-to-helix</code> script resides?

Comment (riarenas) [reply eng/common/templates/steps/send-to-helix.yml:33]: YAML is a weird thing. While removing duplication is always good, we've in the past opted for not adding more indirection to already hard to read templates. We'd definitely consider a PR to see how things look though.

Comment (trylek) [reply eng/common/templates/steps/send-to-helix.yml:33]: Fair enough. This is the original change I made in my futile attempt to make the relevant YAML scripts somewhat easier to read:

https://github.com/dotnet/runtime/pull/42038

I can easily tweak it by moving the inner script around (along with one more runtime-specific variant of <code>send-to-helix-step.yml</code>); if you however believe that it will get rejected on the grounds of YAML script coding standards in the arcade repo (I have no reason to dispute), I guess I'll just revert the runtime change - in fact maybe I don't even need to as my changes will get overwritten with the next arcade ingestion.

Comment (riarenas) [reply eng/common/templates/steps/send-to-helix.yml:33]: >I can easily tweak it by moving the inner script around (along with one more runtime-specific variant of send-to-helix-step.yml)

The `inner` template seems to have some runtime specific steps (the restoring blob tasks), so those shouldn't be part of it. I would also advice against adding any repository specific templates to `eng/common`. As they will bloat other unrelated repositories. The rest of the change looks good to me to help with the duplication, but I would ask @chcosta on whether he thinks adding the inner template is worth it.

>in fact maybe I don't even need to as my changes will get overwritten with the next arcade ingestion.

Yep, the next arcade update will just overwrite any differences with what Arcade has.
